### PR TITLE
[BUG] Conflict in copy: File without extension

### PR DESCRIPTION
**Pull request by Aitorbp**
## Related Issues
App: https://github.com/owncloud/android/issues/4222

- [x] Added changelog files for the fixed issues in folder changelog/unreleased. More info [here](https://github.com/owncloud/android/tree/master/changelog#create-changelog-items)
_____

### CODE REVIEW

**Comment by Aitorbp** [comment]
The `startsWith` function will not work with files that lack an extension for the following reason:
If we have a file called `test` and we want to copy this same file to the same folder. The file will behave as follows:

`“test (1)”.startsWith(“test”)`

This function will return `true` and therefore enter the `INVALID_COPY_INTO_DESCENDANT` exception, since the prefix is the same.

The behavior with extension would be the following:

`“test (1).log”.startsWith(“test.log”)`

The` test.log` prefix is not contained in the target therefore it will return `false`. With a folder it has the same behavior, only instead of having an extension it has a slash `/` at the end.

I remove the startsWith condition from the `CopyRemoteFileOperation` class and keep this logic in the `CopyFileUseCase`, where these exceptions are being handled.

File: owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/files/CopyRemoteFileOperation.kt [hunk 1]
```diff
@@ -68,9 +68,6 @@ class CopyRemoteFileOperation(
             // nothing to do!
             return RemoteOperationResult(ResultCode.OK)
         }
-        if (targetRemotePath.startsWith(sourceRemotePath) && sourceSpaceWebDavUrl == targetSpaceWebDavUrl) {
-            return RemoteOperationResult(ResultCode.INVALID_COPY_INTO_DESCENDANT)
-        }
```
Comment (JuancaG05) [inline owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/files/CopyRemoteFileOperation.kt:73]: Ok, but in the use case we don't have the second part of the condition (`sourceSpaceWebDavUrl == targetSpaceWebDavUrl`), and that's why we cannot copy into descendants in different spaces. The solution would be merging this condition in the use case. But that will be addressed in the new created issue:
- https://github.com/owncloud/android/issues/4293

**Review by JuancaG05 — CHANGES_REQUESTED** [review]
Check the comments about the calens file, and go ahead with the other issue after finishing this one

File: changelog/unreleased/4294 [hunk 2]
```diff
@@ -0,0 +1,7 @@
+Bugfix: Fixed conflict in copy with file without extension
+
+The startsWith function have been removed from CopyRemoteFileOperation because
+it does not work with files that lack an extension. The functionality is handling in CopyFileUseCase.
```
[the remark this answers is not in the document]
Comment (Aitorbp) [reply changelog/unreleased/4294]: Done

File: changelog/unreleased/4294 [hunk 3]
```diff
@@ -0,0 +1,7 @@
+Bugfix: Fixed conflict in copy with file without extension
```
[the remark this answers is not in the document]
Comment (Aitorbp) [reply changelog/unreleased/4294]: Done

**Review by JuancaG05 — APPROVED** [review]
LGTM

**Comment by jesmrec** [comment]
Fixed

**Comment by jesmrec** [comment]
Dont merge till https://github.com/owncloud/android/pull/4295 is also approved

**Pull request merged**